### PR TITLE
fix issues in mac

### PR DESCRIPTION
**Pull request by mary-dcouto**

### CODE REVIEW

File: Ziti.NET.Standard/src/OpenZiti/Native/nAPI.cs [hunk 1]
```diff
@@ -345,9 +345,13 @@ public struct ziti_options {
         //public IntPtr config_types;
         public IntPtr /*public char**/ config_types;
 
-        public Int32 api_page_size;
-
+        public Int32 api_page_size;
```
Comment (dovholuknf) [inline Ziti.NET.Standard/src/OpenZiti/Native/nAPI.cs]: this should just be changed to 'int' or 'uint'. i'm shocked i didn't hit this issue

File: Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs [hunk 2]
```diff
@@ -169,8 +169,12 @@ public class ZitiIdentity {
 			Native.ziti_options ziti_opts = new Native.ziti_options {
 				//app_ctx = GCHandle.Alloc(InitOpts.ApplicationContext, GCHandleType.Pinned),
 				config = InitOpts.IdentityFile,
-				config_types = cfgs,
-				refresh_interval = refreshInterval,
+				config_types = cfgs,
+#if ZITI_X64
+				refresh_interval = (ulong)refreshInterval,
```
Comment (dovholuknf) [inline Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs]: when refresh_interval is set to in this won't be an issue i don't expect?

Comment (mary-dcouto) [reply Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs]: Changed the data type of refreshInterval to long

File: Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs [hunk 3]
```diff
@@ -169,8 +169,12 @@ public class ZitiIdentity {
 			Native.ziti_options ziti_opts = new Native.ziti_options {
 				//app_ctx = GCHandle.Alloc(InitOpts.ApplicationContext, GCHandleType.Pinned),
 				config = InitOpts.IdentityFile,
-				config_types = cfgs,
-				refresh_interval = refreshInterval,
+				config_types = cfgs,
```
Comment (dovholuknf) [inline Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs:172]: all these architecture issues are sorted now, right? we can remove these changes?

File: TestProject/TestProject.csproj [hunk 4]
```diff
@@ -19,6 +19,10 @@
     <PlatformTarget>x86</PlatformTarget>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(OS)' != 'Windows_NT' AND '$(Platform)'=='x64'">
+    <DefineConstants>ZITI_X64;MAC;LINUX</DefineConstants>
```
Comment (dovholuknf) [inline TestProject/TestProject.csproj]: i don't think we need these

File: Samples/Util/ConsoleHelper.cs [hunk 5]
```diff
@@ -41,17 +42,20 @@ public static void AllowAsciEscapeCodes()
...
         /// For whatever reason - writing perfectly fine bytes out to the console using c# is overly difficult.
         /// This method will convert the bytes to output, read lines from that string and write the strings out
         /// line by line. Why _this_ seems to be necessary is still unknown but this does seem to work.
         /// </summary>
         /// <param name="utf8bytes"></param>
         public static void OutputResponseToConsole(byte[] utf8bytes)
-        {
-            AllowAsciEscapeCodes();
+        {
+#if !MACOS && !LINUX
```
Comment (dovholuknf) [inline Samples/Util/ConsoleHelper.cs]: these should be using `RuntimeInformation.IsOSPlatform(OSPlatform.OSX)` and `RuntimeInformation.IsOSPlatform(OSPlatform.Linux)`

File: Samples/Util/ConsoleHelper.cs [hunk 6]
```diff
@@ -10,8 +10,9 @@ public class ConsoleHelper
         // credit for most of this code to https://gist.github.com/tomzorz/6142d69852f831fb5393654c90a1f22e
         private const int STD_OUTPUT_HANDLE = -11;
         private const uint ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004;
-        private const uint DISABLE_NEWLINE_AUTO_RETURN = 0x0008;
-
+        private const uint DISABLE_NEWLINE_AUTO_RETURN = 0x0008;
+
+#if !MACOS && !LINUX
```
Comment (dovholuknf) [inline Samples/Util/ConsoleHelper.cs]: was this a problem if `GetConsoleMode` wasn't called? is it still needed if not called? i'd like to not have to have these if we don't need to

File: Ziti.NET.Standard/src/OpenZiti/Native/nAPI.cs (hunk 1 above)
Comment (mary-dcouto) [reply Ziti.NET.Standard/src/OpenZiti/Native/nAPI.cs]: done

File: Samples/Util/ConsoleHelper.cs (hunk 6 above)
Comment (mary-dcouto) [reply Samples/Util/ConsoleHelper.cs]: you are right, this does not cause any compilation errors, so removing the if condition

File: Samples/Util/ConsoleHelper.cs (hunk 5 above)
Comment (mary-dcouto) [reply Samples/Util/ConsoleHelper.cs]: done

File: TestProject/TestProject.csproj (hunk 4 above)
Comment (mary-dcouto) [reply TestProject/TestProject.csproj]: removed

File: Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs (hunk 3 above)
Comment (mary-dcouto) [reply Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs:172]: refresh interval has to be long in 64 bit processor

Comment (dovholuknf) [reply Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs:172]: the C Sharp sdk can provide an int here and not cast to long. If someone needs a refresh interval > 2billion seconds they can file a pull request :) 

You can remove the if/else and (long) cast, right? c# will automatically fit the incoming c# into into the long

File: Samples/Console.csproj [hunk 7]
```diff
@@ -15,7 +15,12 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>x64</PlatformTarget>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(OS)' != 'Windows_NT'">
+    <PlatformTarget>x64</PlatformTarget>
+    <DefineConstants>MAC;LINUX</DefineConstants>
```
Comment (dovholuknf) [inline Samples/Console.csproj]: i thought we were removing these?

File: Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs (hunk 3 above)
Comment (mary-dcouto) [reply Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs:172]: When I change the data type of refresh_interval to int, it is failing in MAC

File: Samples/Console.csproj (hunk 7 above)
Comment (mary-dcouto) [reply Samples/Console.csproj]: I had removed it from another csproj earlier. Now I removed it from this file as well

File: Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs (hunk 3 above)
Comment (dovholuknf) [reply Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs:172]: It'd be helpful to know how it fails. At runtime, at compile time, or something else? i'll see if i can replicate

Comment (mary-dcouto) [reply Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs:172]: It is failing at runtime. I run this code using DEBUG | 64 or Release | 64 options in MAC.

**Pull request merged**